### DISCUSSION TITLE
Generating inclined grain in 2D

### DISCUSSION THREAD

**Discussion by asmaaharfoush** [neperfepx/neper#185] [Q&A] [answered]
Hello, I am wondering if neper allows me to have a restriction on the inclination angle of the grain on the horizontal axis of the picture produced, for instance, if I need 20% of the grain has no inclination, 20% of the grain inclined 40 degrees, 20% of the grains inclined 80 degrees, 20% of the grain inclined 120 degrees, and 20% of the grain inclined 120 degrees. Is that an option?

Also, I am wondering how I can calculate the grain size if I am going to use any distribution?

Thank you, really appreciate your help!

**Comment by rquey** [comment]
An image would help.

**Comment by asmaaharfoush** [comment, marked as answer]
![inclinded grains](https://user-images.githubusercontent.com/37200158/114861295-e90a6480-9da1-11eb-804f-1f7b93fa7a35.png)

Thanks for your reply this is what I am looking for, I want that 20% of the grain has zero inclination on the horizontal axis, 20%has 40-degree inclination on the horizontal axis, and so on.
There is the arrangement that I need in developing the gain. Is this an option that Neper can provide? if my question is still not clear please let me know.

Thank you!

**Reply by rquey** [reply]
There is no metric for this in Neper, although it could be added (not be me on the short term).

**Reply by asmaaharfoush** [reply]
Thank you!

**Reply by asmaaharfoush** [reply]
Sorry for bothering you again, is there an option in Neper to make all the grain inclined 45 on the horizontal axes?

Thanks!

**Reply by rquey** [reply]
```
$ neper -T -n 1000 -dim 2 -domain "square(1,1):rotate(0,0,1,45):scale(0.5,2,1)" -transform "scale(2,0.5,1),rotate(0,0,1,-45)"
$ neper -V n1000-id1.tess -datacellcol id -cameraangle 12 -imagesize 600:600 -dataedgerad 0.001 -print img
```
![image](https://user-images.githubusercontent.com/2659376/116554483-85cd1600-a8fb-11eb-8781-8214d546f8c6.png)

**Reply by asmaaharfoush** [reply]
This is really helpful, Thanks!
